bijection-avro/src/main/scala/com/twitter/bijection/avro/AvroCodecs.scala

```diff
@@ -109,6 +109,18 @@ object SpecificAvroCodecs {
   }
 
   /**
+   * Returns Injection capable of serializing and deserializing a compiled avro record using org.apache.avro.io.BinaryEncoder.
+   * Accepts a schema to support generated Scala case classes.
+   * @tparam T compiled Avro record
+   * @return Injection
+   */
+  def toBinaryWithSchema[T <: SpecificRecordBase: ClassTag](schema: Schema): Injection[T, Array[Byte]] = {
+    val writer = new SpecificDatumWriter[T](schema)
+    val reader = new SpecificDatumReader[T](schema)
+    new BinaryAvroCodec[T](writer, reader)
```

Comment:
`.getSchema` is part of `SpecificRecordBase` and the empty constructor required by the spec. The unit tests also run successfully.

The difference with `toBinary` is I'd pass in the schema to `SpecificDatum{Writer,Reader}`. If you don't, it will fail at runtime because Avro's `SpecificData` class can't find the SCHEMA$ field (Scala fields are always private). It's all about this stupid Java/Scala difference + reflection.

I've also opened a ticket with Avro directly, but I'm not sure they'll change this only because of Scala.